sbibm/algorithms/abcpy/abcpy_utils.py

```diff
@@ -0,0 +1,124 @@
+from typing import Callable
+import torch
+import numpy as np
+from abcpy.continuousmodels import Continuous, InputConnector
+from abcpy.distances import Euclidean, LogReg, PenLogReg
+from abcpy.output import Journal
+from abcpy.probabilisticmodels import ProbabilisticModel
+from abcpy.statistics import Statistics
+
+
+class ABCpyPrior(ProbabilisticModel, Continuous):
+    def __init__(self, task, name='ABCpy_prior'):
+        self.prior_forward = task.get_prior()
+        self.dim_parameters = task.dim_parameters
+        self.name = task.name if task.name is not None else name
+
+        input_parameters = InputConnector.from_list([])
+        super(ABCpyPrior, self).__init__(input_parameters, self.name + "_prior")
+
+    def forward_simulate(self, abcpy_input_values, num_forward_simulations, rng=np.random.RandomState()):
+        result = np.array(self.prior_forward(num_forward_simulations))
+        # print(num_forward_simulations, result.shape)
```

Comment:
Can be removed

Reply:
Do you mean removing the comment in line 22?

Reply:
I've removed that